src/xtgeo/common/sys.py

```diff
@@ -380,133 +415,136 @@ def splitext(self, lower=False):
 
         return stem, suffix
 
-    def get_cfhandle(self):  # was get_handle
-        """Get SWIG C file handle for CXTgeo.
+    def get_cfhandle(self) -> int:
+        """
+        Get SWIG C file handle for CXTGeo.
 
         This is tied to cfclose() which closes the file.
 
         if _cfhandle already exists, then _cfhandlecount is increased with 1
 
+        Returns:
+            int indicating the file handle number.
+
         """
-        # differ on Linux and other OS as Linux can use fmemopen() in C
-        islinux = True
-        fobj = None
-        if plfsys() != "Linux":
-            islinux = False
+        # Windows and pre-10.13 macOS lack fmemopen()
+        islinux = platform.system() == "Linux"
 
         if self._cfhandle and "Swig Object of type 'FILE" in str(self._cfhandle):
             self._cfhandlecount += 1
-            logger.info("Get SWIG C fhandle no %s", self._cfhandlecount)
+            logger.debug("Get SWIG C fhandle no %s", self._cfhandlecount)
             return self._cfhandle
 
-        if isinstance(self._file, io.BytesIO) and self._mode == "rb" and islinux:
-            fobj = self._file.getvalue()  # bytes type in Python3, str in Python2
-
-            # note that the typemap in swig computes the length for the buf/fobj!
-            self._memstream = True
-
-        elif isinstance(self._file, io.BytesIO) and self._mode == "wb" and islinux:
-            fobj = bytes()
-            self._memstream = True
+        fobj: Union[bytes, str, io.BytesIO, io.StringIO] = self.name
+        if isinstance(self.file, io.BytesIO):
```

Comment:
I think some additional testing should be done here around these blocks when `self.file` is a StringIO. Since it will be a string and not a byte buffer writes can behave differently (i.e., they can break).